### PR TITLE
fix(loggers): add `best` and `latest` aliases to wandb artifact in `WandbLogger`

### LINKED ISSUE #16979: WandbLogger model artifacts do not have `best` aliases when using `save_top_k`> 0
### Bug description

When using WandbLogger and ModelCheckpoint with `save_top_k` > 0, the model artifacts on Wandb do not have `best` aliases -- instead they only show `best_k` and `latest` aliases for the last model artifact that was uploaded to WandB. Note that this artifact version may not be the actual best, but only one of the `best_k`.

### How to reproduce the bug

As per the reproduction code below, this happens because `_scan_checkpoints` override the `best` model tag with `best_k`. 

The below code is an extension of `test_scan_checkpoints` `tests/tests_pytorch/loggers/test_logger.py`

```python
from pathlib import Path
import random
import tempfile
from unittest.mock import patch

from lightning.pytorch.loggers.utilities import _scan_checkpoints

checkpoint_callback_mock = patch("lightning.pytorch.callbacks.ModelCheckpoint")
save_top_k = 5
# Randomly assign one of the checkpoints to be the best
best_i = random.randint(0, max(save_top_k - 1, 0))
with tempfile.TemporaryDirectory() as tmpdir:
    best_k_models = {}
    for i in range(save_top_k):
        ckpt_path = Path(tmpdir) / f"{i}.ckpt"
        ckpt_path.write_text("")
        best_k_models[ckpt_path] = i
        if i == best_i:
            checkpoint_callback_mock.best_model_path = ckpt_path
            checkpoint_callback_mock.best_model_score = i
    checkpoint_callback_mock.best_k_models = best_k_models
    logged_model_time = {}
    checkpoints = _scan_checkpoints(checkpoint_callback_mock, logged_model_time)

    # This passes because the number of checkpoints is correct
    assert len(checkpoints) == save_top_k

    if save_top_k > 0:
        # This fails because the tag is now `best_k`
        assert checkpoints[best_i][-1] == "best"
```

### Error messages and logs

```
assert checkpoints[best_i][-1] == "best"
AssertionError
```

### Environment

<details>
  <summary>Current environment</summary>
I used the latest code on master from this commit: b2b6449be7b50973ab3f1d7fce6d72cb96c3b75f

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

In its current form it is unclear to me what is the usefulness of the `best_k` alias altogether when logging to Wandb: as per the Wandb conventions, the tags/aliases are unique for an artifact in the sense that two different versions of the same artifact can not have the same tag (if ModelCheckpoint outputted an ordering of checkpoints based on the score instead, then tags `best_1`, `best_2`, `best_3`, etc, could be useful).

As a possible solution to resolve this bug, I think resorting back to the following logic makes sense:

```python
     aliases = ["latest", "best"] if p == checkpoint_callback.best_model_path else ["latest"]
```

cc @awaelchli @morganmcg1 @borisdayma @scottire @parambharat

### PR DESCRIPTION
**Pull request by parambharat**
## What does this PR do?

Fixes #16979
Logs the `best` and `latest` tags when model artifacts are logged with `lightning.pytorch.callbacks.ModelCheckpoint`

Here's a sample [wandb workspace ](https://wandb.ai/parambharat/wandb-lightning)that has the corresponding [artifacts.](https://wandb.ai/parambharat/wandb-lightning/artifacts)
Here's a [colab](https://colab.research.google.com/drive/1jx3KmLgrsxjxHPv88Ek6PjPJvAah-lvV?usp=sharing) to reproduce the runs in the workspace above.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you write any **new necessary tests**? (not for typos and docs)
- [X] Did you verify new and **existing tests pass** locally with your changes?

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Comment by awaelchli** [comment]
Hmm, the bot is not triggering the merge 🤔
